libs/src/services/discoveryProvider/index.js

```diff
@@ -28,15 +28,26 @@ class DiscoveryProvider {
 
   async init () {
     let endpoint
+    let pick
+
     if (this.autoselect) {
       endpoint = await this.autoSelectEndpoint()
     } else {
       if (!this.whitelist || this.whitelist.size === 0) {
         throw new Error('Must pass autoselect true or provide whitelist.')
       }
 
-      const pick = _.sample([...this.whitelist]) // selects random element from list.
-      const isValid = await this.ethContracts.validateDiscoveryProvider(pick)
+      // use this as a lookup between health_check endpoint and base url
+      const whitelistMap = {}
+      ;[...this.whitelist].map(url => {
```

Comment:
lgtm, nit, you could probably use some lodash fn here instead of spreading
I think _.each might work? It would be o(n) instead of o(2n)

Reply:
i could also just iterate over the set with a for loop. good ol vanillajs

Reply:
make sure to not do a 

for(let i; i< x; i++) loop. those are blocking and will chunk up the UI on initial load

Reply:
what about for...of?